File: algorithms/codeforces/Alice Bob and Chocolate.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> t(n);
    for (int i = 0; i < n; i++) {
        cin >> t[i];
    }

    int Alice = 0;
    int Bob = 0;
    int l = 0;
    int r = n - 1;
    while (l <= r) {
        if (Alice <= Bob) {
            Alice += t[l];
            l++;
        }
        else {
            Bob += t[r];
            r--;
        }
    }    

    cout << l << " " << n - l;

    return 0;
}

File: algorithms/python/intermediate/algorithmic_complexity/Alice Bob and Chocolate.py
n = int(input())
a = list(map(int, input().split()))

Alice = 0
Bob = 0
l = 0
r = n - 1
while l <= r:
    if Alice <= Bob:
        Alice += a[l]
        l += 1
    elif Alice > Bob:
        Bob += a[r]
        r -= 1
        
print(l, n - l)